### PR TITLE
Adding more comments to the CodePush class

### PR DESCRIPTION
**Pull request by lostintangent**
This PR makes a few minor coding style changes, but mostly adds a bunch of method comments to the `CodePush` class to help document it's internal/JS-public API surface a little better for other contributors.

### CODE REVIEW

File: CodePush.m [hunk 1]
```diff
@@ -93,15 +109,24 @@ - (void)checkForPendingUpdate:(BOOL)needsRestart
             if ([pendingHash isEqualToString:currentHash]) {
                 int rollbackTimeout = [pendingUpdate[PendingUpdateRollbackTimeoutKey] intValue];
                 [self initializeUpdateWithRollbackTimeout:rollbackTimeout needsRestart:needsRestart];
-                
-                // Clear the pending update and sync
-                [preferences removeObjectForKey:PendingUpdateKey];
-                [preferences synchronize];
+            } else {
+                // NOTE: We shouldn't ever reach here
```
Comment (geof90) [inline CodePush.m]: Maybe we should log an inconsistency error instead of an empty else block

**Comment by geof90** [comment]
small comment and LGTM otherwise

File: CodePush.h [hunk 2]
```diff
@@ -1,7 +1,18 @@
 #import "RCTBridgeModule.h"
 
-@interface CodePush : NSObject<RCTBridgeModule>
-
+@interface CodePush : NSObject <RCTBridgeModule>
+
+/*
+ * This method is used to retreive the URL for the most recent
```
Comment (silhouettes) [inline CodePush.h]: retrieve

File: CodePush.h [hunk 3]
```diff
@@ -1,7 +1,18 @@
...
+ * version of the JavaScript bundle. This could be either the
+ * bundle that was packaged with the app binary, or the bundle
+ * that was downloaded as part of a CodePush update. The value returned
+ * should be used to "bootstrap" the React Native bridge.
+ *
+ * This method assumes that your JS bundle is named "main.jsbundle"
+ * and therefore, if it isn't, you should use either the bundleURLForResource:
+ * or bundleURLForResource:withExtension: methods to override thsat behavior.
```
Comment (silhouettes) [inline CodePush.h]: that

File: CodePush.m [hunk 4]
```diff
@@ -61,14 +62,21 @@ + (NSURL *)bundleURLForResource:(NSString *)resourceName
...
     return documentsDirectory;
 }
 
-// Internal API methods
+// Private API methods
+
+/*
+ * This method cancels the currently running rollback
+ * timer, which has the effect of stopping an automatic
+ * rollback from occuring.
```
Comment (silhouettes) [inline CodePush.m]: occurring

File: CodePush.m [hunk 5]
```diff
@@ -76,6 +84,14 @@ - (void)cancelRollbackTimer
     });
 }
 
+/* 
+ * This method checks to see whether a "pending udpate" has been applied
```
Comment (silhouettes) [inline CodePush.m]: update

File: CodePush.m [hunk 6]
```diff
@@ -76,6 +84,14 @@ - (void)cancelRollbackTimer
...
+ * (e.g. install was called with a non-immediate mode), but the app hasn't
+ * yet been restarted (either naturally or synthentically). If there is one,
```
Comment (silhouettes) [inline CodePush.m]: synthetically (or 'manually' or 'artificially' might be better)

Comment (silhouettes) [inline CodePush.m]: programmatically?

File: CodePush.m [hunk 7]
```diff
@@ -151,6 +182,14 @@ - (instancetype)init
     return self;
 }
 
+/*
+ * This method performs the actual initialization work for a update
```
Comment (silhouettes) [inline CodePush.m]: an update

File: CodePush.m [hunk 8]
```diff
@@ -151,6 +182,14 @@ - (instancetype)init
...
+ * to ensure that the neccessary state is setup, including:
```
Comment (silhouettes) [inline CodePush.m]: necessary

File: CodePush.m [hunk 9]
```diff
@@ -318,6 +405,10 @@ - (void)startRollbackTimer:(int)rollbackTimeout
     });
 }
 
+/*
+ * This method isn't publically exposed via the "react-native-code-push"
```
Comment (silhouettes) [inline CodePush.m]: publicly

File: CodePush.m [hunk 10]
```diff
@@ -326,19 +417,26 @@ - (void)startRollbackTimer:(int)rollbackTimeout
     resolve(@(isFailedHash));
 }
 
+/*
+ * This method isn't publically exposed via the "react-native-code-push"
```
Comment (silhouettes) [inline CodePush.m]: publicly

File: CodePush.m [hunk 11]
```diff
@@ -346,13 +444,18 @@ - (void)startRollbackTimer:(int)rollbackTimeout
     resolve([NSNull null]);
 }
 
-// This function is exposed solely for immediately installed
-// update support, and shouldn't be consumed directly by user code.
-RCT_EXPORT_METHOD(restartImmedidateUpdate:(int)rollbackTimeout)
+/*
+ * This method isn't publically exposed via the "react-native-code-push"
```
Comment (silhouettes) [inline CodePush.m]: publicly

**Comment by silhouettes** [comment]
Just a bunch of typos, sorry for being nitpicky! LGTM

**Pull request merged**